modules/htlc/internal/keeper/keeper.go

```diff
@@ -40,57 +46,47 @@ func (k Keeper) GetCdc() *codec.Codec {
 }
 
 // CreateHTLC creates an HTLC
-func (k Keeper) CreateHTLC(ctx sdk.Context, htlc types.HTLC, hashLock []byte) (sdk.Tags, sdk.Error) {
+func (k Keeper) CreateHTLC(ctx sdk.Context, htlc types.HTLC, hashLock types.HTLCHashLock) sdk.Error {
 	// check if the hash lock already exists
 	if k.HasHashLock(ctx, hashLock) {
-		return nil, types.ErrHashLockAlreadyExists(types.DefaultCodespace, fmt.Sprintf("the hash lock already exists: %s", hex.EncodeToString(hashLock)))
+		return types.ErrHashLockAlreadyExists(types.DefaultCodespace, fmt.Sprintf("the hash lock already exists: %s", hex.EncodeToString(hashLock)))
 	}
 
-	// transfer the specified tokens to a dedicated HTLC Address
-	if _, err := k.bk.SendCoins(ctx, htlc.Sender, auth.HTLCLockedCoinsAccAddr, htlc.Amount); err != nil {
-		return nil, err
+	// transfer the specified tokens to HTLC module address
+	if err := k.sk.SendCoinsFromAccountToModule(ctx, htlc.Sender, types.ModuleName, htlc.Amount); err != nil {
+		return err
 	}
 
-	// add to coinflow
-	ctx.CoinFlowTags().AppendCoinFlowTag(ctx, htlc.Sender.String(), auth.HTLCLockedCoinsAccAddr.String(), htlc.Amount.String(), sdk.CoinHTLCCreateFlow, "")
-
 	// set the HTLC
```

Comment:
add Event for `SendCoinsFromAccountToModule`

Reply:
all bank sending logic has events.